jdbc/src/main/java/com/interface21/jdbc/core/JdbcTemplate.java

```diff
@@ -1,17 +1,81 @@
 package com.interface21.jdbc.core;
 
+import com.interface21.jdbc.core.preparedstatement.PreparedStatementInitializer;
+import com.interface21.jdbc.core.preparedstatement.PreparedStatementSpecification;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
 
-public class JdbcTemplate {
+public record JdbcTemplate(DataSource dataSource) {
 
     private static final Logger log = LoggerFactory.getLogger(JdbcTemplate.class);
 
-    private final DataSource dataSource;
+    public void execute(final PreparedStatementSpecification specification) {
+        try (
+                Connection connection = dataSource.getConnection();
+                PreparedStatement preparedStatement = PreparedStatementInitializer.initialize(
+                        connection,
+                        specification
+                )
+        ) {
+            preparedStatement.executeUpdate();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
```

Comment:
라이브러리를 만드는 미션이니 명세도 최대한 자세하게 작성해 보는 경험을 해 봐도 좋을 것 같아요.
- javadoc
- throws 

```suggestion
    public void execute(final PreparedStatementSpecification specification) throws RuntimeException {
        try (
                Connection connection = dataSource.getConnection();
                PreparedStatement preparedStatement = PreparedStatementInitializer.initialize(
                        connection,
                        specification
                )
        ) {
            preparedStatement.executeUpdate();
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
```